hexa/pipelines/models.py

```diff
@@ -188,6 +192,10 @@ def last_run(self) -> "PipelineRun":
 
     def upload_new_version(self, user: User, zipfile, parameters):
         if self.last_version:
+            if self.schedule and parameters:
+                raise InvalidDefinition(
```

Comment:
In this case you should raise a more specific exception, like `PipelineDoesNotSupportParametersError`. Otherwise how would you know later which error code to return?

Reply:
The other option would be to have an InvalidDefinitionError exception class with an error code enum but I don't see the added value compared to an explicit exception class.

Reply:
Indeed, you're right.I'll go with the first option.